src/daemon/sigfetcher.rs

```diff
@@ -110,14 +114,29 @@ fn get_sigs<C: secp256k1::Verification>(
             pubkey,
             tx.txid()
         );
-        if let Err(e) = tx.add_signature(0, pubkey.key, sig, secp) {
+        if let Err(e) = tx.add_signature(pubkey.key, sig, secp) {
             // FIXME: should we loudly fail instead ? If the coordinator is sending us bad
             // signatures something shady's happening.
             log::error!("Error while adding signature for presigned tx: '{}'", e);
             continue;
         }
     }
 
+    if let Some(our_stk_key) = our_stk_key {
+        if !contains_our_signature {
+            // Oh, the coordinator didn't have our signature. Here it is!
+            if let Some(our_sig) = tx.signatures().remove(&our_stk_key) {
```

Comment:
Can we even get into a state where this returns `None`? Since asserting here 3 days to release sounds scary, can you add a `log::error` in the `else` if we should always have it?

EDIT: i was completely wrong, it's more than possible: it's usual. Leaving the comment for hysterical raisins.